upgrade-2.x.sh

```diff
@@ -249,6 +250,176 @@ function device_type_match {
+function pre_update_pi_bootfiles_removal {
+    local boot_files_for_removal=('start_db.elf' 'fixup_db.dat')
+    for f in "${boot_files_for_removal[@]}"; do
+        echo "Removing $f from boot partition"
+        rm -f "/mnt/boot/$f"
+    done
+    sync
+}
+
+function pre_update_fix_bootfiles_hook {
+    log "Applying bootfiles hostapp-hook fix"
+    local bootfiles_temp
+    bootfiles_temp=$(mktemp)
+    curl -f -s -L -o "$bootfiles_temp" https://raw.githubusercontent.com/resin-os/resinhup/master/upgrade-patches/0-bootfiles || log ERROR "Couldn't download fixed '0-bootfiles', aborting."
+    chmod 755 "$bootfiles_temp"
+    mount --bind "$bootfiles_temp"  /etc/hostapp-update-hooks.d/0-bootfiles
+}
+
+function hostapp_based_update {
+    local update_package=$1
+    local storage_driver
+    local sysroot="/mnt/sysroot/inactive"
+
+    case ${SLUG} in
+        raspberry*)
+            log "Running pre-update fixes for ${SLUG}"
+            pre_update_pi_bootfiles_removal
+            if ! version_gt "${VERSION_ID}" "2.7.6" ; then
+                pre_update_fix_bootfiles_hook
+            fi
+            ;;
+        *)
+            log "No device-specific pre-update fix for ${SLUG}"
+    esac
+
+    if ! [ -S "/var/run/docker-host.sock" ]; then
+        ## Happens on devices booting after a regular HUP update onto a hostapps enabled resinOS
+        log "Do not have docker-host running"
+        log "Clean inactive partition"
+        rm -rf "${sysroot:?}/"*
+        log "Getting storage driver info"
+        storage_driver=$(cat /boot/storage-driver)
+        log "Starting docker-host with ${storage_driver} storage driver"
+        dockerd --log-driver=journald -s="${storage_driver}" --data-root="${sysroot}/docker" -H unix:///var/run/docker-host.sock --pidfile=/var/run/docker-host.pid --exec-root=/var/run/docker-host --bip 10.114.101.1/24 --fixed-cidr=10.114.101.128/25 --iptables=false &
+        sleep 10
+    else
+        if [ -f "$sysroot/resinos.fingerprint" ]; then
+            # Happens on a device, which has HUP'd from a non-hostapp resinOS first
+            # then updated again but this time with hostapp-update. The previous "active"
+            # partition now inactive, and still has leftover data
+            log "Have docker-host running, with dirty inactive partition"
+            systemctl stop docker-host
+            log "Clean inactive partition"
+            rm -rf "${sysroot:?}/"*
+            systemctl start docker-host
+            sleep 10
+        fi
+    fi
+
+    log "Starting hostapp-update"
+    hostapp-update -i "${update_package}" || log ERROR "hostapp-update has failed..."-
+}
+
+function non_hostapp_to_hostapp_update {
+    local update_package=$1
+    local sysroot="/mnt/sysroot/inactive"
+    local temp_sysroot
+
+    find_partitions
+    stop_services
+
+    # Mount spare root partition
+    umount "${update_part}" || true
+    mkfs.ext4 -F -L "${update_label}" "${update_part}"
```

Comment:
Use the same parameters we use in resinOS when creating partitions: https://github.com/resin-os/meta-resin/blob/b32f8be8cd90d39732c707f009213e616a6396fa/meta-resin-common/recipes-containers/mkfs-hostapp-native/files/create#L16

Reply:
Done